Feat: User details page

DevdouglasGfs opened the pull request:
# User Page
This page is dedicated to show user data.

## Steps to access
To access this page have 2 ways:
1. accessing the route `/users/user/:id` proving the user ID;
2. or when in the `/users` route, click at a user in the list provided;

It solves the issue #7

DevdouglasGfs committed:
Chore: Add styled components to project deps

DevdouglasGfs committed:
Feat: Create common components
Create common components which will be used in the project;

DevdouglasGfs committed:
Styles: Change global styling
Add new fonts (IBM PLEX MONO, Inter);
Change default background and text color;

DevdouglasGfs committed:
Chore: Remove unnecessary stylesheets

DevdouglasGfs committed:
Ref: Recreate login page using styled components
Rebuilt utilizing components created with styled-components

DevdouglasGfs committed:
Ref: Recreate user management  page
Rebuilt utilizing components created with styled-components

DevdouglasGfs committed:
Ref: Recreate users list page
Rebuilt utilizing components created with styled-components

DevdouglasGfs committed:
Ref: Recreate CreateUser component
Rebuilt utilizing components created with styled-components

DevdouglasGfs committed:
Ref: Recreate base route /users
Rebuilt nested components using styled-components

DevdouglasGfs committed:
Chore: Update routing structure
Refined the routing structure for improved organization

DevdouglasGfs committed:
Chore: Custom react hook useGetUsers
useGetUsers will generate a Error if it do not receive arguments

DevdouglasGfs committed:
Feat: Create User page to show user informations

DevdouglasGfs committed:
Fix: list elements in Table overflowing

DevdouglasGfs committed:
Fix: Send the user to the correct page
After of a new user is created:
The user should be redirected to the users page
When a error occurs:
In the console, the error should be displayed with the error object

DevdouglasGfs commented:
# Screen Presentation
Below, you can see how the page appears. 


![Screenshot_20240626-054408~2](https://github.com/DevdouglasGfs/taqtile/assets/105404079/55078b4a-9161-4093-8d82-b88e7f4f2680)
![Screenshot_20240626-055450](https://github.com/DevdouglasGfs/taqtile/assets/105404079/a6315707-dcf0-4464-b51d-34b8960c0994)

DevdouglasGfs committed:
Ref: Remove unused component

racsandoval commented on src/pages/users/user/index.tsx:
```diff
@@ -0,0 +1,81 @@
+import styled from "styled-components";
+import { Spinner } from "../../../components/common/spinner";
+
+export interface UserPage {
```
pra que serve essa interface?

DevdouglasGfs replied:
Para  nada, removi ela agora
Pensei que a usava em outros lugares, mas apenas a exportava

racsandoval reviewed (commented):
muitos arquivos de estilização mas nenhum print 😢

DevdouglasGfs committed:
Chore: extends link props
Also make flex prop receive a syring

DevdouglasGfs committed:
Chore: Use semantic tags

DevdouglasGfs committed:
Feat: Add navigation buttons

DevdouglasGfs committed:
Ref: remove unused interface UserPage

DevdouglasGfs committed:
Chore: remove unused interface

DevdouglasGfs committed:
Chore: Rename styledForm to Form

DevdouglasGfs committed:
Chore: Make private custom properties

DevdouglasGfs committed:
Chore: Improve accessibility and code quality
Remove unnecessary code and improve accessibility

DevdouglasGfs committed:
Chore: Move default theme to another folder

DevdouglasGfs committed:
Chore: Rename component files to PascalCase
Updated component filenames to use PascalCase

DevdouglasGfs committed:
Feat: Improve components reusabiliy and maintain
Do not use strictly the values gone from props;
( take these from the default theme if they are not provided )

DevdouglasGfs committed:
Chore: Remove unused file

racsandoval commented on src/components/common/Input.tsx:
```diff
@@ -0,0 +1,44 @@
+        return ".2px solid rgba(255, 255, 255, .2)"
+    }};
+    padding: ${props => {
+        if (props.$size === "small") { return "4px 8px" }
+        else if (props.$size === "medium") { return "8px 16px" }
+        else if (props.$size === "large") { return "12px 24px" }
+        else return "8px 16px"
```
como ele sempre retorna se passa num `if`, não precisa dos `else`s

DevdouglasGfs replied:
Vou alterar!

racsandoval commented on src/components/create-user/index.tsx:
```diff
@@ -30,18 +45,17 @@ export default function CreateUser({ open, setOpen }: { open: boolean, setOpen:
-            if (!dialog.current?.contains(target as Node)) onClose()
+            if (!modal.current?.contains(target as Node)) onClose()
         })
     })
```
esse `useEffect` está sem array de dependências, portanto ele roda em todas as renderizações

DevdouglasGfs replied:
Eu não tenho certeza se resolvi esse problema com a nova alteração, mas fiz as alteração como me foi sugerido no console

racsandoval commented on src/components/create-user/index.tsx:
```diff
@@ -30,18 +45,17 @@ export default function CreateUser({ open, setOpen }: { open: boolean, setOpen:
     // eslint-disable-next-line
     const onClickOutside = useEffect(() => {
         document.addEventListener('mousedown', ({ target }) => {
-            if (!dialog.current?.contains(target as Node)) onClose()
+            if (!modal.current?.contains(target as Node)) onClose()
         })
```
poderia limpar o listener quando o `useEffect` rodasse novamente

DevdouglasGfs replied:
Vou fazer isso!

DevdouglasGfs replied:
Apliquei o clear

racsandoval commented on src/pages/users/index.tsx:
```diff
@@ -1,21 +1,56 @@
+    </>
+}
 
 export function UsersPage() {
```
me parece estranho a página de usuários cuidar do roteamento da aplicação

DevdouglasGfs replied:
Vou alterar o roteamento

DevdouglasGfs replied:
Acabei por alterar o roteamento na PR #22 @racsandoval está bem? Pode ser assim?

racsandoval commented on src/pages/users/index.tsx:
```diff
@@ -1,21 +1,56 @@
+        <ThemeProvider theme={theme}>
+            <Routes>
+                <Route path="/" element={<>
+                    <main>
+                        <Hero>
+                            <Heading>
+                                <span>Dashboard</span>
+                                Taqtile
+                            </Heading>
+                        </Hero>
+                        <Outlet />
+                    </main>
```
esse jsx poderia estar em outro componente

DevdouglasGfs replied:
Vou cria-lo!

DevdouglasGfs committed:
Chore: Replace else if with if statements
else if may cause multiple returns in styled-components and cause errors

DevdouglasGfs committed:
Fix: Modal height trying to get the height of the content

DevdouglasGfs committed:
Fix: DOM re-renderizing every time

DevdouglasGfs committed:
Style: Improve the appearance and fix issue with Input component

DevdouglasGfs committed:
Fix: else statement without return

DevdouglasGfs committed:
Feat: Create Header component

DevdouglasGfs committed:
Merge branch 'feature/add-users-page' into feature/user-details-page

DevdouglasGfs committed:
Style: Update color and background color default in the project

DevdouglasGfs committed:
Feat: Add $accentElementDIr property to Heading component

DevdouglasGfs committed:
Style: Change accent color to Input component

DevdouglasGfs committed:
Ref: Split the logic and remove the unnecessary code

DevdouglasGfs committed:
Feat: integrate Label and Input components in the Field component

DevdouglasGfs committed:
Chore: Update routing strategy

DevdouglasGfs committed:
Chore: Export pages adhering a common style

DevdouglasGfs committed:
Ref: Split the UI and the logic of validation

DevdouglasGfs committed:
Feat: Create a arrow function to validate if the birth date is valid

DevdouglasGfs committed:
Merge branch 'feature/add-users-page' into feature/user-details-page

DevdouglasGfs committed:
Chore: remove code repetting

DevdouglasGfs committed:
Chore: remove code repetting

DevdouglasGfs committed:
Chore: Remove duplicated components
